### ISSUE TITLE
Issue with addition of new speakers

### ISSUE THREAD

**Issue by RamakrishnaChaitanya** [AI4Bharat/Indic-TTS#37]
Hi, I'm trying to finetune the "en+hi" model on the openSLR dataset. It looks like the model was trained using 4 speakers. However, I would like to use a new speaker instead of the available speakers. So, i have formatted the custom dataset accordingly with new speaker labels. And in addition, I have changed the number of speakers to 5 in the config file after adding an ID for the new speaker in the .json file. I finetuned the weights with this information for 100 epoch's. Later, i just tried to do inference using the new checkpoint. However, I'm getting the following error

size mismatch for emb_g.weight: copying a param with shape torch.Size([4, 512]) from checkpoint, the shape in current model is torch.Size([5, 512]).

From this, it looks like the finetuned model still somehow correlates with 4 speakers, but not 5 speakers. Is my understanding correct? Please let me know the procedure to resolve this issue. @GokulNC @svp19 

![Image](https://github.com/user-attachments/assets/e355c40d-44bc-42af-bec2-fe36f3522b13)

**Comment by RamakrishnaChaitanya** [comment]
I loaded the wrong checkpoint. Issue got resolved after loading the right checkpoint
